helper/helper_test.go

```diff
@@ -1,12 +1,174 @@
+	return nil
+}
+
+func (r *restoreReaderTestImpl) copyData(num int64) (int64, error) {
+	return 1, nil
+}
+
+func (r *restoreReaderTestImpl) copyAllData() (int64, error) {
+	return 1, nil
+}
+
+func (r *restoreReaderTestImpl) closeFileHandle() {
+}
+
+func (r *restoreReaderTestImpl) getReaderType() ReaderType {
+	return "nil"
+}
+
+type helperTestStep struct {
+	getRestorePipeWriterArgExpect string
+	getRestorePipeWriterResult    bool
+	checkSkipFileArgTableOid      int
+	checkSkipFileResult           bool
+}
+
+type restoreMockHelperImpl struct {
+	stepNo           int
+	expectedOidBatch []oidWithBatch
+	expectedSteps    []helperTestStep
+
+	openedPipesMap map[string]string // Ginkgo matcher works over map value, will diplicate key here
+	restoreData    *restoreReaderTestImpl
+}
+
+func (h *restoreMockHelperImpl) openedPipes() []string {
+	if h.openedPipesMap == nil {
+		h.openedPipesMap = make(map[string]string)
+
+		for k := range pipesMap {
+			h.openedPipesMap[k] = k
+		}
+	}
+	ret := make([]string, 0, len(h.openedPipesMap))
+	for k := range h.openedPipesMap {
+		ret = append(ret, k)
+	}
+	return ret
+}
+
+func (h *restoreMockHelperImpl) getCurStep() helperTestStep {
+	Expect(h.stepNo).To(BeNumerically("<", len(h.expectedSteps)))
+	return h.expectedSteps[h.stepNo]
+}
+
+func (h *restoreMockHelperImpl) closeAndDeletePipe(tableOid int, batchNum int) {
+}
+
+func newHelperTest(batches []oidWithBatch, steps []helperTestStep) *restoreMockHelperImpl {
+	var ret = new(restoreMockHelperImpl)
+	ret.expectedOidBatch = batches
+	ret.expectedSteps = steps
+	ret.openedPipesMap = nil
+	ret.restoreData = &restoreReaderTestImpl{}
+
+	return ret
+}
+
+func (h *restoreMockHelperImpl) getOidWithBatchListFromFile(oidFileName string) ([]oidWithBatch, error) {
+	return h.expectedOidBatch, nil
+}
+
+func (h *restoreMockHelperImpl) checkForSkipFile(pipeFile string, tableOid int) bool {
+	step := h.getCurStep()
+	Expect(tableOid).To(Equal(step.checkSkipFileArgTableOid))
+	ret := h.getCurStep().checkSkipFileResult
+	return ret
+}
+
+func (h *restoreMockHelperImpl) createPipe(pipe string) error {
+	// Check that pipe was not opened yet
+	Expect(h.openedPipes()).ShouldNot(ContainElement(pipe))
+
+	h.openedPipesMap[pipe] = pipe
+	return nil
+}
+
+func (h *restoreMockHelperImpl) flushAndCloseRestoreWriter(pipeName string, oid int) error {
+	// Check that we are closing pipe which is opened
+	Expect(h.openedPipes()).To(ContainElement(pipeName))
+	delete(h.openedPipesMap, pipeName)
+	return nil
+}
+
+func (h *restoreMockHelperImpl) getRestoreDataReader(fileToRead string, objToc *toc.SegmentTOC, oidList []int) (IRestoreReader, error) {
+	if h.restoreData != nil {
+		return h.restoreData, nil
+	}
+	return nil, errors.New("getRestoreDataReader Not implemented")
+}
+
+func (h *restoreMockHelperImpl) getRestorePipeWriter(currentPipe string) (*bufio.Writer, *os.File, error) {
+	h.stepNo++
+	Expect(currentPipe).To(Equal(h.getCurStep().getRestorePipeWriterArgExpect))
```

Comment:
maybe something like
```suggestion
	step := h.getNextStep()
	Expect(currentPipe).To(Equal(step.getRestorePipeWriterArgExpect))
```